### ISSUE TITLE
MaterialX materials with 1x1 images are not rendered correctly in HdStorm

### ISSUE THREAD

**Issue by pablode** [PixarAnimationStudios/USD#2140]
### Description of Issue

The following example shows MaterialX and UsdPreviewSurface materials referencing 1x1 and 2x2 images of the color red:
![comparison](https://user-images.githubusercontent.com/3663466/210078538-26edb3b6-19e2-4401-ab85-6472acea8f7b.png)

The 1x1 MaterialX case is not rendered correctly. I've furthermore verified that it is rendered as red in MaterialXView.

### Steps to Reproduce
1. Open following model with usdview: [SimplePoC.zip](https://github.com/PixarAnimationStudios/USD/files/10325057/SimplePoC.zip)
2. Observe that the image is red, but the rectangle is black
3. Resize the image to 2x2 or greater
4. The rectangle now has the correct red color

### System Information (OS, Hardware)
Windows, RTX 2060

### Package Versions
USD v22.11

### Build Flags
`--materialx --build-variant release --opencolorio --openimageio`

**Comment by sunyab** [comment]
Filed as internal issue #USD-7838

**Comment by klucknav** [comment]
This is actually due to the MaterialX shaders. The shader code generated for texture nodes start with a [check that the width is strickly greater than 1](https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/libraries/stdlib/genglsl/mx_image_color3.glsl#L5). So every texture that is 1xH will get the defaultval. While this value can be specified on the texture node, it defaults to black for color3 texture nodes. This is why we are seeing black for the 1x1 MaterialX case. I actually did see black when I opened the material with the 1x1 texture in the MaterialX Viewer as well. However, I am told that the MaterialX example applications resize 1x1 textures to be 4x4 to pass that size check and not use the defaultval. So perhaps that is why you were able to see the texture color.
There is a [GitIssue](https://github.com/AcademySoftwareFoundation/MaterialX/issues/1438) on the MaterialX repo to add handing for 1x1 textures without applications having to resize them.
